packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx

```diff
@@ -1,58 +1,116 @@
+        return [styles.root, styles.openPopover, styles.numberDirtyFields];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ theme, ownerState }) => css`
+        position: relative;
+        cursor: pointer;
+        display: flex;
+        background-color: ${theme.palette.common.white};
+        border-color: ${theme.palette.grey[100]};
+        border-radius: 2px;
+
+        &:hover {
+            border-color: ${theme.palette.grey[100]};
+            background-color: ${theme.palette.common.white};
+        }
+
+        &:focus {
+            border-color: ${theme.palette.primary.main};
+            background-color: ${theme.palette.common.white};
+        }
+
+        ${ownerState.openPopover &&
+        css`
+            border-color: ${theme.palette.primary.main};
+        `}
+
+        ${ownerState.numberDirtyFields &&
+        css`
+            border-color: ${theme.palette.grey[400]};
+            font-weight: ${theme.typography.fontWeightBold};
+
+            &:disabled {
+                border-color: ${theme.palette.grey[100]};
+            }
+        `}
+    `,
+);
+
+const FilterBadge = styled("span", {
+    name: "CometAdminFilterBarButton",
+    slot: "filterBadge",
+    overridesResolver(_, styles) {
+        return [styles.filterBadge];
+    },
+})(
+    css`
+        margin-left: 6px;
+    `,
+);
+
+export interface FilterBarButtonProps extends ThemedComponentBaseProps<{ root: typeof Button; filterBadge: "span" }>, ButtonProps {
     dirtyFieldsBadge?: React.ComponentType<FilterBarActiveFilterBadgeProps>;
     numberDirtyFields?: number;
     openPopover?: boolean;
 }
 
-const FilterBarButton = ({
-    children,
-    dirtyFieldsBadge,
-    numberDirtyFields,
-    openPopover,
-    classes,
-    endIcon = <ChevronDown />,
-    className,
-    ...buttonProps
-}: FilterBarButtonProps & WithStyles<typeof styles>): React.ReactElement => {
+export function FilterBarButtonWithStyles(inProps: FilterBarButtonProps) {
+    const {
+        children,
+        dirtyFieldsBadge,
+        numberDirtyFields,
+        openPopover,
+        classes,
+        endIcon = <ChevronDown />,
+        className,
+        slotProps,
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminFilterBarButton",
+    });
+
+    const ownerState: OwnerState = {
+        openPopover,
+        numberDirtyFields,
+    };
+
     const hasDirtyFields = !!(numberDirtyFields && numberDirtyFields > 0);
     const FilterBarActiveFilterBadgeComponent = dirtyFieldsBadge ? dirtyFieldsBadge : FilterBarActiveFilterBadge;
 
     return (
-        <Button
-            className={clsx(className, classes.root, hasDirtyFields && classes.hasDirtyFields, openPopover && classes.open)}
-            disableRipple
-            endIcon={endIcon}
-            variant="outlined"
-            {...buttonProps}
-        >
+        <Root ownerState={ownerState} {...restProps} {...slotProps?.root} disableRipple endIcon={endIcon} variant="outlined">
```

Comment:
Please add the `restProps` and `slotProps` to the end, so they can override the other values.